src/components/Terminal.tsx

```diff
@@ -0,0 +1,250 @@
+		background-color: #000;
+		overflow-y: scroll;
+		cursor: default;
+		position: absolute;
+		right: 0;
+		left: 0;
+		top: 0;
+		bottom: 0;
+	}
+
+	.xterm .xterm-screen {
+		position: relative;
+	}
+
+	.xterm .xterm-screen canvas {
+		position: absolute;
+		left: 0;
+		top: 0;
+	}
+
+	.xterm .xterm-scroll-area {
+		visibility: hidden;
+	}
+
+	.xterm .xterm-char-measure-element {
+		display: inline-block;
+		visibility: hidden;
+		position: absolute;
+		left: -9999em;
+	}
+
+	.xterm.enable-mouse-events {
+		/* When mouse events are enabled (eg. tmux), revert to the standard pointer cursor */
+		cursor: default;
+	}
+
+	.xterm:not(.enable-mouse-events) {
+		cursor: text;
+	}
+`;
+
+const TtyContainer = styled(Box)`
+	position: relative;
+	height: 100%;
+
+	${defaultXtermStyle}
+
+	.xterm-viewport::-webkit-scrollbar-track {
+		background-color: transparent
+		border-bottom-right-radius: 0;
+	}
+
+	.xterm-viewport::-webkit-scrollbar {
+		width: 10px;
+		background-color: transparent;
+		border-radius: 0;
+	}
+
+	.xterm-viewport::-webkit-scrollbar-thumb {
+		background-color: #e9e9e9;
+	}
+`;
+
+const TtyInner = styled.div`
+	position: absolute;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+`;
+
+interface ThemedTerminalProps extends TerminalProps {
+	theme: Theme;
+}
+
+class Terminal extends React.Component<ThemedTerminalProps, {}> {
+	private tty: Xterm;
+	// Used as the element ID to mount XTERM into
+	private mountElementId: string;
+
+	constructor(props: ThemedTerminalProps) {
+		super(props);
+
+		const termConfig = _.assign({}, props.config, {
+			cols: 80,
+			cursorBlink: false,
+			rows: 24,
+			useStyle: true,
+			fontFamily: Theme.monospace,
+			lineHeight: 1.4,
+			theme: {
+				background: '#343434',
+				cursor: props.nonInteractive ? '#343434' : undefined,
+			},
+		});
+
+		// Allow an existing tty instance to be rebound to this react element
+		this.tty = this.props.ttyInstance || new Xterm(termConfig);
+
+		this.mountElementId = `terminal-${uuid.v1()}`;
+
+		this.resize = this.resize.bind(this);
+	}
+
+	componentDidMount() {
+		this.open();
+
+		// Wait before writing to the xterm instance, so that the term can be sized to the screen correctly
+		setTimeout(() => {
+			// If this is a non-interactive terminal, rebind <ctrl+c> to copy command
+			if (this.props.nonInteractive && document.execCommand) {
+				this.tty.attachCustomKeyEventHandler(e => {
+					// Ctrl + C
+					if (e.ctrlKey && e.keyCode === 67) {
+						document.execCommand('copy');
+						return false;
+					}
+					// to make the typings happy
+					return true;
+				});
+			}
+
+			window.addEventListener('resize', this.resize);
+
+			this.resize();
+		}, 100);
+	}
+
+	componentWillUnmount() {
+		if (this.tty) {
+			window.removeEventListener('resize', this.resize);
+			this.tty.destroy();
+		}
+	}
+
+	destroy() {
+		window.removeEventListener('resize', this.resize);
+
+		if (!this.props.persistent) {
```

Comment:
`persistent` is always used because you've passed in a `ttyInstance` that you don't want destroyed, is that right?

Might be nice to remove this parameter, and just use `!this.props.ttyInstance` here instead - if you pass in an instance you're always responsible for cleaning it up, if you don't then it always gets cleaned up for you.

Reply:
yep agreed